scripts/evaluate_performance.jl

```diff
@@ -1,24 +1,52 @@
-# run this script as `julia evaluate_performance.jl res.bson` or `julia evaluate_performance.jl dir`
-# in the second case, it will recursively search for all compatible files in subdirectories
-target = ARGS[1]
-
 using DrWatson
 @quickactivate
 using EvalMetrics
 using FileIO
 using BSON
 using DataFrames
+using PrettyTables
+using PrettyTables.Crayons
+using Statistics
+using Base.Threads: @threads
+
+# pkgs which come from BSONs
 using ValueHistories
 
+
+"""
+	_prefix_symbol(prefix, s)
+
+Modifies symbol `s` by adding `prefix` with underscore.
+"""
+_prefix_symbol(prefix, s) = Symbol("$(prefix)_$(String(s))")
+
+
+"""
+	compute_stats(f::String)
+
+Computes evaluation metrics from the results of experiment in serialized bson at path `f`.
+Returns a DataFrame row with metrics and additional metadata for groupby's.
+Hash of the model parameters is precomputed in order to make the groupby easier.
+"""
 function compute_stats(f::String)
-	data = load(f)
-	println(abspath(f))
-	#map(p->println("$(p[1]) = $(p[2])"), collect(pairs(data[:parameters])))
-	scores_labels = [(data[:val_scores], data[:val_labels]), (data[:tst_scores], data[:tst_labels])]
-	setnames = ["validation", "test"]
+	r = load(f)
+	row = (
+		modelname = r[:modelname],
+		dataset = r[:dataset],
+		phash = hash(r[:parameters]),
+		parameters = savename(r[:parameters]), 
+		seed = r[:seed])
+	
+	if Symbol("anomaly_class") in keys(r)
+		row = merge(row, (anomaly_class = r[:anomaly_class]))
+	end
 
-	results = []
-	for (scores, labels) in scores_labels
+	for splt in ["val", "tst"]
+		scores = r[_prefix_symbol(splt, :scores)]
+		labels = r[_prefix_symbol(splt, :labels)]
+
+		any(isnan.(scores)) && error("$(splt)_scores contain NaN")
```

Comment:
Can we filter out the scores and labels that contain NaN?

Reply:
Yes we can, however I would not go as far as filtering out all NaNs as oftentimes you are left with just an empty array. I am not sure if we should tolerate some NaNs given some threshold. Are 4 NaNs a sign of corrupted model/results or is it 5? Who is here to say, that the methods just does not purposefully fail on so many occasions such that the performance seems better.
This is maybe too harsh of a limit as some of the methods such as `ABOD` on `kdd, mammography, miniboone` produce at least on NaN for each hyperparameter, yielding no evaluation files for this dataset/model combination.

Reply:
I would put the limit probably at one half.